modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/GossipDaemon.scala

```diff
@@ -40,115 +39,127 @@ object GossipDaemon {
-          .semiflatTap(_ => metrics.updateRumorsConsumed("success", signedRumor))
-          .getOrElseF {
-            logger.debug(s"Unhandled rumor ${signedRumor.value.show} with hash ${hash.show}") >>
-              metrics.updateRumorsConsumed("unhandled", signedRumor)
-          }
-          .handleErrorWith { err =>
-            logger.error(err)(s"Error handling rumor ${signedRumor.value.show} with hash ${hash.show}") >>
-              metrics.updateRumorsConsumed("error", signedRumor)
           }
-    }
 
-    private def runGossipRounds: F[Unit] =
-      for {
-        _ <- Temporal[F].sleep(cfg.interval)
-        activeHashes <- rumorStorage.getActiveHashes
-        seenHashes <- rumorStorage.getSeenHashes
-        peers <- clusterStorage.getPeers
-        selectedPeers <- Random[F].shuffleList(peers.toList).map(_.take(cfg.fanout))
-        _ <- selectedPeers.parTraverse { peer =>
-          runGossipRound(activeHashes, seenHashes, peer).handleErrorWith { err =>
-            logger.error(err)(s"Error running gossip round with peer ${peer.show}") >>
-              Spawn[F].start(healthcheck.triggerCheckForPeer(peer)).void
-          }
+        private def sortRumors(batch: RumorBatch): RumorBatch =
+          batch.filter { case (_, s) => s.value.isInstanceOf[CommonRumorBinary] } ++
+            batch.filter { case (_, s) => s.value.isInstanceOf[PeerRumorBinary] }
+              .sortBy(_._2.asInstanceOf[Signed[PeerRumorBinary]].ordinal)
+
+        private def handleRumor(har: HashAndRumor): F[Unit] = har match {
+          case (hash, signedRumor) =>
+            rumorHandler
+              .run((signedRumor.value, nodeId))
+              .semiflatTap(_ => metrics.updateRumorsConsumed("success", signedRumor))
+              .getOrElseF {
+                logger.debug(s"Unhandled rumor ${signedRumor.value.show} with hash ${hash.show}") >>
+                  metrics.updateRumorsConsumed("unhandled", signedRumor)
+              }
+              .handleErrorWith { err =>
+                logger.error(err)(s"Error handling rumor ${signedRumor.value.show} with hash ${hash.show}") >>
+                  metrics.updateRumorsConsumed("error", signedRumor)
+              }
         }
-      } yield ()
-
-    private def runGossipRound(activeHashes: List[Hash], seenHashes: List[Hash], peer: Peer): F[Unit] =
-      for {
-        startRequest <- StartGossipRoundRequest(activeHashes).pure[F]
-        startResponse <- gossipClient.startGossiping(startRequest).run(peer)
-        inquiry = startResponse.offer.diff(seenHashes)
-        answer <- rumorStorage.getRumors(startResponse.inquiry)
-        endRequest = EndGossipRoundRequest(answer, inquiry)
-        _ <- metrics.updateRumorsSent(answer)
-        endResponse <- gossipClient.endGossiping(endRequest).run(peer)
-        _ <- rumorQueue.offer(endResponse.answer)
-        _ <- metrics.updateRumorsReceived(endResponse.answer)
-      } yield ()
+
+        private def selectPeers: F[Unit] =
```

Comment:
Actual changes in this file start from here (above there are changes mostly in the line indentation).